src/Effects.js

```diff
@@ -1445,6 +1445,40 @@ module.exports = class Effects {
         }
       }
     };
+
+    this.music_wave = {
+      wave: [],
+      init: function () {
+        let wave = {
+          inc: 360/15,
+          heightFactor: 1,
+          yHalf: 300,
+          colorUnit: 400,
+          colorHalf: 200,
+        };
+        this.wave.push(wave);
+      },
+
+      draw: function (context) {
+        const centroid = context.centroid;
+        let scale = p5.map(centroid, 5000, 8000, 0, 1);
+        let a = 0;
+        p5.background('black');
+        const wave = this.wave[0];
+        for (let i = 1; i <= wave.colorUnit; i+=5) {
+          p5.stroke(lerpColorFromPalette(i/wave.colorUnit));
+          let yInitial = wave.yHalf - Math.abs(wave.yHalf * scale * (wave.heightFactor/wave.colorHalf) * p5.cos(a));
```

Comment:
Style suggestion: If we extract `Math.abs(wave.yHalf * scale * (wave.heightFactor/wave.colorHalf) * p5.cos(a)` into it's own variable, it will be easier to update in the future and, in my opinion, the code becomes more readable.

Reply:
I made an `amplitude` variable for it.